pandas/tests/io/excel/test_writers.py

```diff
@@ -1385,6 +1385,18 @@ def test_to_excel_empty_frame(self, engine, ext):
             expected = DataFrame()
             tm.assert_frame_equal(result, expected)
 
+    def test_to_excel_raising_warning_when_cell_character_exceed_limit(
+        self, path, engine
+    ):
+        # GH#56954
+        df = DataFrame({"A": ["a" * 32768]})
+        msg = "Cell contents too long, truncated to 32767 characters"
+        with tm.assert_produces_warning(
+            UserWarning, match=msg, raise_on_extra_warnings=False
```

Comment:
What extra warning is raised here?

Reply:
From commit https://github.com/pandas-dev/pandas/pull/57103/commits/babdf29c0420a4093adc96b7c174155209b4ae85, 

the check https://github.com/pandas-dev/pandas/actions/runs/7678882560/job/20929191559 rise `pandas/tests/io/excel/test_writers.py::TestExcelWriter::test_to_excel_raising_warning_when_cell_character_exceed_limit[openpyxl-.xlsx] - AssertionError: Caused unexpected warning(s): [('EncodingWarning', EncodingWarning("'encoding' argument not specified"), '/home/runner/micromamba/envs/test/lib/python3.10/site-packages/openpyxl/writer/excel.py', 199)]` 

and check https://github.com/pandas-dev/pandas/actions/runs/7678882560/job/20929192274 rasie `pandas/tests/io/excel/test_writers.py::TestExcelWriter::test_to_excel_raising_warning_when_cell_character_exceed_limit[openpyxl-.xlsm] - AssertionError: Caused unexpected warning(s): [('DeprecationWarning', DeprecationWarning('datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).'), '/home/runner/micromamba/envs/test/lib/python3.12/site-packages/openpyxl/packaging/core.py', 99), ('DeprecationWarning', DeprecationWarning('datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).'), '/home/runner/micromamba/envs/test/lib/python3.12/site-packages/openpyxl/writer/excel.py', 292)]`

`pandas/tests/io/excel/test_writers.py::TestExcelWriter::test_to_excel_raising_warning_when_cell_character_exceed_limit` is what we added in this pr. I just wrote the test following the other's in same file. If there is a better way to avoid these extra warnings, I am very happy to change codes!

Reply:
Gotcha. Yeah since these are external warnings this is fine for now